src/society_stage/StrategicCamera.cs

```diff
@@ -137,6 +138,35 @@ public bool PanCamera(float delta, float forwardMovement, float leftRightMovemen
         return true;
     }
 
+    [RunOnKey("g_pan_mouse", CallbackRequiresElapsedTime = false)]
+    public bool PanCameraWithMouse(float delta)
+    {
+        if (!Visible)
+            return false;
+
+        if (mousePanningStart == null)
+        {
+            mousePanningStart = CursorWorldPos;
+        }
+        else
+        {
+            var movement = mousePanningStart.Value - CursorWorldPos;
+            ApplyPan(movement * CameraPanSpeed * ZoomLevel * ZoomLevelMovementSpeedModifier * delta / -10);
+        }
+
+        return false;
+    }
+
+    [RunOnKeyUp("g_pan_mouse")]
+    public bool ReleasePanCameraWithMouse()
+    {
+        if (!Visible)
```

Comment:
Shouldn't the mouse pan still be reset even if this is not visible?
Also eating the mouse up event is rarely a good idea.